cli/src/main/java/com/devonfw/tools/ide/tool/GlobalToolCommandlet.java

```diff
@@ -116,29 +115,29 @@ protected boolean isExtract() {
   }
 
   @Override
-  public ToolInstallation install(boolean silent, VersionIdentifier configuredVersion, ProcessContext processContext, Step step) {
+  protected ToolInstallation doInstall(ToolInstallRequest request) {
 
+    VersionIdentifier resolvedVersion = request.getRequested().getResolvedVersion();
     if (this.context.getSystemInfo().isLinux()) {
       // on Linux global tools are typically installed via the package manager of the OS
       // if a global tool implements this method to return at least one PackageManagerCommand, then we install this way.
       List<PackageManagerCommand> commands = getInstallPackageManagerCommands();
       if (!commands.isEmpty()) {
-        boolean newInstallation = runWithPackageManager(silent, commands);
-        Path rootDir = getInstallationPath(getConfiguredEdition(), configuredVersion);
-        return createToolInstallation(rootDir, configuredVersion, newInstallation, processContext, false);
+        boolean newInstallation = runWithPackageManager(request.isSilent(), commands);
+        Path rootDir = getInstallationPath(getConfiguredEdition(), resolvedVersion);
+        return createToolInstallation(rootDir, resolvedVersion, newInstallation, request.getProcessContext(), request.isAdditionalInstallation());
       }
     }
 
-    ToolEdition toolEdition = getToolWithEdition();
-    Path installationPath = getInstallationPath(toolEdition.edition(), configuredVersion);
+    ToolEdition toolEdition = getToolWithConfiguredEdition();
+    Path installationPath = getInstallationPath(toolEdition.edition(), resolvedVersion);
     // if force mode is enabled, go through with the installation even if the tool is already installed
     if ((installationPath != null) && !this.context.isForceMode()) {
-      return toolAlreadyInstalled(silent, toolEdition, configuredVersion, processContext, false);
+      return toolAlreadyInstalled(request);
     }
     String edition = toolEdition.edition();
     ToolRepository toolRepository = this.context.getDefaultToolRepository();
-    VersionIdentifier resolvedVersion = toolRepository.resolveVersion(this.tool, edition, configuredVersion, this);
```

Comment:
This was a bug that the `resolvedVersion` was only computed here but above `configuredVersion` was already used for package manager that may not support our IDEasy version patterns.